src/receiver.ts

```diff
@@ -113,33 +114,43 @@ export default class Receiver extends EventEmitter {
                     try {
                         const response: Response = await this.fetch(item.url)
 
-                        // parse the notification
+                        const responseText = await response.text()
                     
-                        const jsonldParser = JsonLdParser.fromHttpResponse(
-                            response.url,
-                            response.headers.get('content-type') || "application/ld+json"
-                        )
-
-                        // transform bodystream
-                        //const bodyStream = new ReadableWebToNodeStream(response.body || new ReadableStream())
-
-                        // TODO: Fix this when NodeJS vs. Stream API chaos is over
-                        const bodyStream = new Readable()
-                        bodyStream.push(await response.text())
-                        bodyStream.push(null)
-
-                        // parse the notification
-                        const notification = await EventNotification.parse(bodyStream, jsonldParser)
-
-                        // emit an event with notification
-                        const idToCheck = strategy == 'notification_id' ? item.url : notification.id.value
-
-                        if (this.db && !this.db.get(idToCheck)) {
-                            this.emit('notification', notification)
-                            this.db.set(idToCheck, true)
+                        try {
+                            // parse the notification
+                            const jsonldParser = JsonLdParser.fromHttpResponse(
+                                response.url,
+                                response.headers.get('content-type') || "application/ld+json"
+                            )
+
+                            // transform bodystream
+                            //const bodyStream = new ReadableWebToNodeStream(response.body || new ReadableStream())
+
+                            // TODO: Fix this when NodeJS vs. Stream API chaos is over
+                            const bodyStream = new Readable()
+                            bodyStream.push(responseText)
+                            bodyStream.push(null)
+
+                            // parse the notification
+                            const notification = await EventNotification.parse(bodyStream, jsonldParser)
+
+                            // emit an event with notification
+                            const idToCheck = strategy == 'notification_id' ? item.url : notification.id.value
+
+                            if (this.db && !this.db.get(idToCheck)) {
+                                this.emit('notification', notification)
+                                this.db.set(idToCheck, true)
+                            }
+                        }
+                        catch (e) {
```

Comment:
This is kind of a brute force way :) Wouldn't be better to check the `response.headers.get('content-type')` and ignore everything but json-ld? With everything else, an error would be expected behavior. 

In any case, I have to provide parsers for the other RDF serializations as well.

Reply:
Anything can fail, servers and data can be nasty. If something fails the code now says : "ok, you are nasty, i saw this content, I will remember that you have sent it and will ignore it for the future.